infrastructure/version/src/main/java/tech/pegasys/teku/infrastructure/version/VersionProvider.java

```diff
@@ -44,21 +45,41 @@ public static Bytes32 getDefaultGraffiti() {
     }
   }
 
+  public static String defaultStoragePath() {
+    final String detectedOS = normalizeOS(normalize("os.name"));
+    return defaultStoragePathForNormalizedOS(detectedOS, System.getenv());
+  }
+
+  private static String getImplementationVersion() {
+    final String version = VersionProvider.class.getPackage().getImplementationVersion();
+    return version != null ? version : "<Unknown>";
+  }
+
   private static String detectOS() {
     final String detectedOS = normalizeOS(normalize("os.name"));
     final String detectedArch = normalizeArch(normalize("os.arch"));
     return detectedOS + '-' + detectedArch;
   }
 
-  public static String detectJvm() {
+  private static String detectJvm() {
     final String detectedVM = normalizeVM(normalize("java.vendor"), normalize("java.vm.name"));
     final String detectedJavaVersion = System.getProperty("java.specification.version");
     return detectedVM + "-java-" + detectedJavaVersion;
   }
 
-  public static String defaultStoragePath() {
-    final String detectedOS = normalizeOS(normalize("os.name"));
-    return defaultStoragePathForNormalizedOS(detectedOS, System.getenv());
+  private static Optional<String> getCommitHash() {
+    try (InputStream is =
+        VersionProvider.class.getClassLoader().getResourceAsStream("git.properties")) {
+      if (is != null) {
+        final Properties properties = new Properties();
+        properties.load(is);
+        return Optional.ofNullable(properties.getProperty("git.commit.id"));
+      } else {
+        return Optional.empty();
+      }
+    } catch (IOException ex) {
+      throw new UncheckedIOException(ex);
+    }
```

Comment:
i think we should structure this so that we can pass the inputStream, and actually test it...